main.py

```diff
@@ -422,6 +422,40 @@ def command_skip() -> None:
     active_voice_client.stop()
 
 
+async def finish_bust() -> None:
+    """End the current bust."""
+    global active_voice_client
+    global original_bot_nickname
+    global current_channel_content
+    global current_channel
+    global total_song_len
+
+    # Disconnect from voice if necessary
+    if active_voice_client and active_voice_client.is_connected():
+        await active_voice_client.disconnect()
+
+    # Restore the bot's original guild nickname (if it had one)
+    bot_member = current_channel.guild.get_member(client.user.id)
+    if original_bot_nickname:
+        await bot_member.edit(nick=original_bot_nickname)
```

Comment:
We will probably have to further factor this out if we fix #68, though I think it's fine keeping it in here for now. Up to you, since the purpose of this commit is largely code reorganization before other commits.

Reply:
Yes, I think it's best to fix that separately, though hopefully such a change would be rather small.